test/e2e/registration_service_test.go

```diff
@@ -487,6 +487,8 @@ func (s *registrationServiceTestSuite) TestPhoneVerification() {
 
 	// Now approve the usersignup.
 	userSignup.Spec.Approved = true
+	//states.SetApproved(userSignup, true)
```

Comment:
@alexeykazakov so it seems that this is possibly not a good place to test the migration.  From what I can gather from the currently failing e2e tests, a phone verification test is failing as the migration is not happening fast enough.  Perhaps it might be better to test the migration somewhere in usersignup_test instead?

Reply:
I doubt the problem is with the speed of the migration. We use `WaitForMasterUserRecord()` to wait for the MUT get provisioned after updating the Approved in the UserSignup spec. The wait timeout should be more than enough.

The test looks correct. So, let's investigate what's happening to make sure there is no bugs in the runtime code.